docs/design/architecture.md

```diff
@@ -170,47 +397,89 @@ The following diagram illustrates the Kata Containers network hotplug workflow.
 
 ## Storage
-Container workloads are shared with the virtualized environment through [virtio-fs](https://virtio-fs.gitlab.io/).
 
-The [devicemapper `snapshotter`](https://github.com/containerd/containerd/tree/master/snapshots/devmapper) is a special case. The `snapshotter` uses dedicated block devices rather than formatted filesystems, and operates at the block level rather than the file level. This knowledge is used to directly use the underlying block device instead of the overlay file system for the container root file system. The block device maps to the top read-write layer for the overlay. This approach gives much better I/O performance compared to using `virtio-fs` to share the container file system.
+### virtio SCSI
+
+If a block-based graph driver is configured,
+`virtio-scsi` is used to "share" the workload image (such as
+`busybox:latest`) into the container's environment inside the VM.
+
+### virtio FS
+
+In all other cases a [`virtio-fs`](https://virtio-fs.gitlab.io)
+(`VIRTIO`) mount point is used instead. The [agent](#agent) uses this
+mount point as the root filesystem for the container processes.
+
+For `virtio-fs`, the [runtime](#runtime) starts one `virtiofsd` daemon
+(that runs in the host context) for each VM created.
+
+### Devicemapper
 
-Kata Containers has the ability to hotplug and remove block devices, which makes it possible to use block devices for containers started after the VM has been launched.
+The
+[devicemapper `snapshotter`](https://github.com/containerd/containerd/tree/master/snapshots/devmapper)
+is a special case. The `snapshotter` uses dedicated block devices
+rather than formatted filesystems, and operates at the block level
+rather than the file level. This knowledge is used to directly use the
+underlying block device instead of the overlay file system for the
+container root file system. The block device maps to the top
+read-write layer for the overlay. This approach gives much better I/O
+performance compared to using `virtio-fs` to share the container file
+system.
 
-Users can check to see if the container uses the devicemapper block device as its rootfs by calling `mount(8)` within the container.  If the devicemapper block device
-is used, `/` will be mounted on `/dev/vda`.  Users can disable direct mounting of the underlying block device through the runtime configuration.
+#### Hot plug and unplug
+
+Kata Containers has the ability to hot plug add and hot plug remove
+block devices. This makes it possible to use block devices for
+containers started after the VM has been launched.
+
+Users can check to see if the container uses the `devicemapper` block
+device as its rootfs by calling `mount(8)` within the container.  If
+the `devicemapper` block device is used, the root filesystem (`/`)
+will be mounted from `/dev/vda`.  Users can disable direct mounting of
+the underlying block device through the runtime
+[configuration](#configuration).
```

Comment:
This goes along with my comment at the start. I am confused on how storage works between the container image that is started in the VM and any mounts that are done to the VM. This section seems to blend between the two.

Reply:
I'm not that comfortable with this part.
Users can check whether a container is using device mapper or not, but usually only admins should change the configuration (and that's also an argument for why we should disable annotations, but that's a different discussion).

Reply:
Section has only been reformatted so let's attack this in a future PR.